### PR TITLE
[network-policy-engine] migrate network-policy-engine to distroless

### PR DESCRIPTION
**Pull request by artbin**
## Description

Network-policy-engine module is based now on a distroless image.

## Why do we need it, and what problem does it solve?

We'd like to base our images on a distroless image.

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: network-policy-engine
type: fix
summary: network-policy-engine module is based on a distroless image.
impact_level: default
```

### CODE REVIEW

**Comment by artbin** [comment]
Manual testing
==============

Documentation screenshot:
Screenshot 2023-11-28 at 17 51 36


Add a NetworkPolicy:
```bash
kubectl apply -f - <<EOF
kind: NetworkPolicy
apiVersion: networking.k8s.io/v1
metadata:
  namespace: d8-system
  name: web-deny-all
spec:
  podSelector:
    matchLabels:
      app: documentation
  ingress: []
EOF
```

Documentation screenshot:
Screenshot 2023-11-28 at 17 52 35

File: modules/050-network-policy-engine/images/kube-router/werf.inc.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      cat <<EOF >> go.mod
+      module github.com/cloudnativelabs/kube-router
+
+      go 1.20
+      EOF
+    - GIT_COMMIT=$(git describe --tags --dirty)
+    - BUILD_DATE=$(date +%Y-%m-%dT%H:%M:%S%z)
+    - GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -mod vendor -ldflags "-X github.com/cloudnativelabs/kube-router/pkg/cmd.version=${GIT_COMMIT} -X github.com/cloudnativelabs/kube-router/pkg/cmd.buildDate=${BUILD_DATE}" -o kube-router cmd/kube-router/kube-router.go
+    - export GOPROXY={{ $.GOPROXY }}
+    - make kube-router
```
Comment (RomanenkoDenys) [inline modules/050-network-policy-engine/images/kube-router/werf.inc.yaml:35]: Why we first build kube-router and than do make kube-router ?

**Review by RomanenkoDenys — COMMENTED** [review]
some question

File: modules/050-network-policy-engine/images/kube-router/werf.inc.yaml (hunk 1 above)
Comment (artbin) [reply modules/050-network-policy-engine/images/kube-router/werf.inc.yaml]: fixed

**Review by name212 — APPROVED** [review]

**Pull request merged**